# **Pandas: Data Correlations**

## Finding Relationships  
A great aspect of Pandas module is the `corr()` method.  

The **corr()**mrthod calculates the the relationship between each column of a Data set.  

We'll be testing it on our Dataset file named 'data.csv'


In [11]:
import pandas as pd

df = pd.read_csv('CSVs/data.csv')

print(df)

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[169 rows x 4 columns]


Now we'll test **corr()** method on out data set, but we must make sure that thier   
are no non-numeric entries as **corr()** works on numeric only data sets.

#### Example: Using corr() on df to check for relations

In [12]:
print(df.corr())

          Duration     Pulse  Maxpulse  Calories
Duration  1.000000 -0.155408  0.009403  0.922717
Pulse    -0.155408  1.000000  0.786535  0.025121
Maxpulse  0.009403  0.786535  1.000000  0.203813
Calories  0.922717  0.025121  0.203813  1.000000


## Explaining Result

The out of **corr()** is a table with bunch of numbers. These numbers represent how well  
the relationship is between two column.  

The numbers varries from -1 to 1.  

- 1 means that there is a 1 to 1 relationship (a perfect correlation), and for this data set, each time a value went up in the first column, the other one went up as well.  

- 0.9 is also a good relationship, and if you increase one value, the other will probably increase as well.  

- -0.9 would be just as good relationship as 0.9, but if you increase one value, the other will probably go down.  

- 0.2 means NOT a good relationship, meaning that if one value goes up does not mean that the other will.



**What is good correlation?** It depends on the use, but in general at least `0.6` (or `-0.6`)  
would be enough to call it a good correlation.

### Perfect Correlation

We can see that the correlation between the same columns is `1.000000`, which is called perfect  
correlation, which also makes sense since both are exactly the same.  

### Good Correlation

'Duration' and 'Calories' got a `0.922721` correlation, which is a very good correlation, and we  
can relate it as the longer you work out the more calories you burn, and vice versa.  

### Bad Correlation

'Duration' and 'Maxpulse' got a `0.009403` correlation, which is a bad correlation, meaning we  
can not predict maxpulse by just looking at duration of the work out, and vice versa.  
